wlm/src/main/java/ai/djl/serving/wlm/LmiUtils.java

```diff
@@ -45,67 +45,101 @@ public final class LmiUtils {
 
     private static final Logger logger = LoggerFactory.getLogger(LmiUtils.class);
 
-    private static final List<String> DEEPSPEED_MODELS =
-            List.of(
-                    "bert",
+    private static final Set<String> LMI_DIST_MODELS =
+            Set.of("falcon", "gpt_neox", "t5", "llama", "mpt", "gpt_bigcode");
+    private static final Set<String> VLLM_MODELS =
+            Set.of(
+                    "aquila",
+                    "baichuan",
                     "bloom",
-                    "gpt_neo",
-                    "gpt_neox",
+                    "chatglm",
+                    "deci",
+                    "falcon",
+                    "gemma",
                     "gpt2",
+                    "gpt_bigcode",
                     "gptj",
+                    "gpt_neox",
+                    "internlm2",
+                    "llama",
+                    "mistral",
+                    "mixtral",
+                    "mpt",
                     "opt",
-                    "roberta",
-                    "xlm-roberta");
+                    "phi",
+                    "qwen",
+                    "qwen2",
+                    "stablelm");
 
     private LmiUtils() {}
 
     static String inferLmiEngine(ModelInfo<?, ?> modelInfo) throws ModelException {
-        // MMS/Torchserve
-        if (Files.isDirectory(modelInfo.modelDir.resolve("MAR-INF"))) {
-            logger.info("Found legacy torchserve model, use Python engine.");
-            return "Python";
-        }
-
-        Properties prop = modelInfo.prop;
+        Properties prop = modelInfo.getProperties();
         HuggingFaceModelConfig modelConfig = getHuggingFaceModelConfig(modelInfo);
+        String features = Utils.getEnvOrSystemProperty("SERVING_FEATURES");
+        String rollingBatch = prop.getProperty("option.rolling_batch", "auto");
         if (modelConfig == null) {
-            String engineName = isTrtLLM(modelInfo) ? "MPI" : "Python";
+            String engineName = isTrtLLM(rollingBatch, features) ? "MPI" : "Python";
             logger.info("No config.json found, use {} engine.", engineName);
             return engineName;
         }
-        String features = Utils.getenv("SERVING_FEATURES");
         String modelType = modelConfig.getModelType();
         String engineName;
         if ("stable-diffusion".equals(modelType)) {
             // TODO: Move this from hardcoded to deduced in PyModel
             prop.setProperty("option.entryPoint", "djl_python.stable-diffusion");
-            engineName = "DeepSpeed";
-        } else if (isDeepSpeedRecommended(modelType)) {
-            engineName = "DeepSpeed";
-        } else if (features != null && features.contains("trtllm")) {
             engineName = "MPI";
+        } else if (isTrtLLM(rollingBatch, features)) {
+            engineName = "MPI";
+        } else if ("lmi-dist".equals(rollingBatch) || isLMIDistRecommended(modelType, features)) {
```

Comment:
regarding to the order. Could we do model specific? Each LMI version will have a recommended model like MPT for LMI-Dist and LLAMA for vLLM?

Reply:
Yea that makes sense - to start we can have map from model type to backend. Maybe in the future (not sure if this will be inside or outside model server) we can further tune the recommendation based on some other factors like input/output max

Reply:
updated to a model level mapping.

I do want to change the structure here a bit, since now we're doing more than inferring the engine. Need to think about it a a bit more and can follow up - we're likely going to have more configs we want to auto set upfront if not provided